kedro/io/cached_dataset.py

```diff
@@ -94,8 +94,8 @@ def _from_config(config: dict, version: Version | None) -> AbstractDataset:
 
     def _describe(self) -> dict[str, Any]:
         return {
-            "dataset": self._dataset._describe(),
-            "cache": self._cache._describe(),
+            "dataset": self._dataset._pretty_repr(self._dataset._describe()),
+            "cache": self._cache._pretty_repr(self._cache._describe()),
```

Comment:
Does that mean we need to update all the datasets?

I expected the other way round and we don't have to couple the dataset implementation with the pretty print.

<img width="600" alt="image" src="https://github.com/kedro-org/kedro/assets/18221871/ecd4db44-e1bf-402b-a522-9a53d76ab837">

Can we add that pretty print method to the base class instead?

Reply:
`_pretty_repr` is already a method of `AbstractDataset`. Could you please clarify what you mean by "to the base class instead"?

Reply:
Alright, I was under the impression the update need to be propagated to all dataset.

Is the name _pretty_repr_ decided on purpose?
It seems that _repr_pretty_ is something IPython support natively. 
https://ipython.readthedocs.io/en/stable/api/generated/IPython.lib.pretty.html

Reply:
Well, I didn't aim to redefine IPython's `_repr_pretty` I needed some method separate from `__repr__` to be able to call it from an inherited class, like in this example. We can rename it if you find it confusing.

Reply:
I was actually thinking opposite, though it's unlikely someone is relying on `repr(catalog)`.

```python
class DataCatalog:
    def __repr_pretty_(self):
          return pprint(self._describe(), ...)
```

This avoid breaking change while adding the nicely formatted printing. `repr` should technically able to construct the object, but this is not a requirement of `_describe`, sometimes deliberate hidden (for credentials, connection str).

Reply:
I see the point about coupling `_describe()` and pretty printing. The reason for that was to keep the format `Class(atr_1=val_1, .., )` for the nested datasets like in this example. I'll think of an alternative approach.

Reply:
After the last changes `_describe()` is decoupled from pretty printing, so `_describe()` doesn’t depend on it